src/merge-exports.js

```diff
@@ -112,6 +115,30 @@ export async function mergeExports(options) {
         logger(`${result.wrappers.length} wrappers created`);
     }
 
+    if (options.preserveExtensions) {
```

Comment:
initially, i had this logic lower down in `create-exports` but it felt more disruptive there, and it was easier to slot in at this level. let me know if there's a better place for it to go

Reply:
ignore the above, this approach had issues whereby successive runs would add more and more `.js` extensions 😅 fixed in later commmits